network-store-server/src/main/java/com/powsybl/network/store/server/NetworkStoreRepository.java

```diff
@@ -704,52 +723,117 @@ public void deleteVoltageLevel(UUID networkUuid, int variantNum, String voltageL
-        return getIdentifiable(networkUuid, variantNum, generatorId, mappings.getGeneratorMappings());
+        Optional<Resource<GeneratorAttributes>> generator = getIdentifiable(networkUuid, variantNum, generatorId, mappings.getGeneratorMappings());
+
+        generator.ifPresent(equipment -> {
+            Map<OwnerInfo, List<ReactiveCapabilityCurvePointAttributes>> reactiveCapabilityCurvePoints = getReactiveCapabilityCurvePoints(networkUuid, variantNum, EQUIPMENT_ID_COLUMN, generatorId);
+            insertReactiveCapabilityCurvePointsInEquipments(networkUuid, List.of(equipment), reactiveCapabilityCurvePoints);
+        });
+        return generator;
     }
 
     public List<Resource<GeneratorAttributes>> getGenerators(UUID networkUuid, int variantNum) {
-        return getIdentifiables(networkUuid, variantNum, mappings.getGeneratorMappings());
+        List<Resource<GeneratorAttributes>> generators = getIdentifiables(networkUuid, variantNum, mappings.getGeneratorMappings());
+
+        Map<OwnerInfo, List<ReactiveCapabilityCurvePointAttributes>> reactiveCapabilityCurvePoints = getReactiveCapabilityCurvePoints(networkUuid, variantNum, EQUIPMENT_TYPE_COLUMN, ResourceType.GENERATOR.toString());
+
+        insertReactiveCapabilityCurvePointsInEquipments(networkUuid, generators, reactiveCapabilityCurvePoints);
+
+        return generators;
     }
 
     public List<Resource<GeneratorAttributes>> getVoltageLevelGenerators(UUID networkUuid, int variantNum, String voltageLevelId) {
-        return getIdentifiablesInVoltageLevel(networkUuid, variantNum, voltageLevelId, mappings.getGeneratorMappings());
+        List<Resource<GeneratorAttributes>> generators = getIdentifiablesInVoltageLevel(networkUuid, variantNum, voltageLevelId, mappings.getGeneratorMappings());
+
+        List<String> equipmentsIds = generators.stream().map(Resource::getId).collect(Collectors.toList());
+
+        Map<OwnerInfo, List<ReactiveCapabilityCurvePointAttributes>> reactiveCapabilityCurvePoints = getReactiveCapabilityCurvePointsWithInClause(networkUuid, variantNum, EQUIPMENT_ID_COLUMN, equipmentsIds);
+
+        insertReactiveCapabilityCurvePointsInEquipments(networkUuid, generators, reactiveCapabilityCurvePoints);
+
+        return generators;
     }
 
     public void updateGenerators(UUID networkUuid, List<Resource<GeneratorAttributes>> resources) {
         updateIdentifiables(networkUuid, resources, mappings.getGeneratorMappings(), VOLTAGE_LEVEL_ID_COLUMN);
+
+        // To update the generator's reactive capability curve points, we will first delete them, then create them again.
+        // This is done this way to prevent issues in case the reactive capability curve point's primary key is to be
+        // modified because of the updated equipment's new values.
+        for (Resource<GeneratorAttributes> resource : resources) {
+            deleteReactiveCapabilityCurvePoints(networkUuid, resource.getVariantNum(), resource.getId());
```

Comment:
Maybe add a function `deleteReactiveCapabilityCurvePoints` which takes a list of resources as parameter instead of this loop to avoid multi SQL requests

Reply:
Done